language-plutus-core/src/Language/PlutusCore/TypeSynthesis.hs

```diff
@@ -39,34 +37,51 @@ isType Type{} = True
 isType _      = False
 
 -- | Create a new 'Type' for an integer operation.
-intop :: MonadQuote m => m (Type TyNameWithKind ())
+intop :: (MonadQuote m) => m (Type TyNameWithKind ())
 intop = do
-    nam <- liftQuote $ freshTyName () ""
+    nam <- newTyName (Size ())
     let ity = TyApp () (TyBuiltin () TyInteger) (TyVar () nam)
         fty = TyFun () ity (TyFun () ity ity)
-    Right t <- runExceptT $ annotateType $ TyForall () nam (Size ()) fty
-    pure t
+    pure $ TyForall () nam (Size ()) fty
 
 -- | Create a new 'Type' for an integer relation
-intRel :: MonadQuote m  => m (Type TyNameWithKind ())
+intRel :: (MonadQuote m)  => m (Type TyNameWithKind ())
```

Comment:
Do we really need these additional `()`?

Reply:
Oh, no. That is a mistake.